lib/block-supports/layout.php

```diff
@@ -63,14 +63,18 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 	if ( $content_size || $wide_size ) {
 		ob_start();
 		?>
+			<?php echo '.wp-container-' . $id; ?> {
+				--wp--layout--content-size: <?php echo $content_size ? $content_size : $wide_size; ?>;
+				--wp--layout--wide-size: <?php echo $wide_size ? $wide_size : $content_size; ?>;
```

Comment:
Last remark I have is about the naming. I don't have any opinion on that, but I just want to make sure this does make sense and is consistent with what we do elsewhere cc @nosolosw 

The other thing I wanted to mention is that "content size" and "wide size" are only defined and useful if the layout is "default" (which is the only layout we support right now, but I'm thinking about flex, absolute, grid layouts later. These have different kind of configs and styles). I guess this last point diminishes the need for these CSS variables because there's no guarantee that they exist.

Reply:
[As far as naming goes](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/), what we have is:

- presets: `--wp--preset--<preset-category>--<preset-slug>`
- custom vars by the theme: `--wp--custom--<variable-name>`
- styles: `--wp--style--color--link`

I'm fine with whatever is consistent and clear for folks, but perhaps a suggestion would be to follow the styles naming `--wp--style--layout-*`.

These styles aren't generated by the theme.json engine, though, but by the block ― so I'm not sure if it's worth making that difference clear or aligning names. One thing I've noticed is that if we do this, they become public API, which so far we've tried to avoid. I haven't been able to follow the layout work closely but by skimming quickly at this code it looks like we attach this code only if a block supports the layout and has the `layout` attribute. I suppose then that there may be situations in which these aren't present (the user didn't select any layout).

Reply:
I'm personally still hesitant about this PR specially because the layout config can change depending on the "type" of the layout, while right now we do support just one type, I expect we'll introduce others pretty soon (horizontal flex, grid, absolute...) and in all of these these two sizes don't make sense.